archunit/src/main/java/com/tngtech/archunit/core/domain/JavaClass.java

```diff
@@ -1372,6 +1419,42 @@ public boolean isAnonymous() {
         return isAnonymousClass();
     }
 
+    private static class Superclass {
+        private static final Superclass ABSENT = new Superclass(Optional.<JavaClass>absent(), Optional.<JavaType>absent());
+
+        private final Optional<JavaClass> rawType;
+        private final Optional<JavaType> genericType;
+
+        private Superclass(Optional<JavaClass> rawType, JavaType genericType) {
+            this(rawType, Optional.of(genericType));
+        }
+
+        private Superclass(Optional<JavaClass> rawType, Optional<JavaType> genericType) {
```

Comment:
I thought about this, the problem is only, that both come from a different source. And for example `signature` is null, if the type and supertype are not generic :thinking: That's why I kept it like this, even though I also had that thought.
What would be possible would be to move the logic further to the importer, e.g. the `JavaClass.Builder`. But it's well possible, that the `rawType` can be derived from the `superclass` string property in `JavaClassProcessor`, but the signature is null and no generic info can be derived :man_shrugging: 
But maybe if the generic type can be derived from the signature, then that should be the single source of truth and we drop the other info. Just that I feel like I can't foresee all the consequences, if we really "overwrite" the original `superclass` property with the erasure from the generic `signature`. On the other hand there is a ton of tests to tell us if it still behaves correctly :thinking: Just not sure about some exotic scenarios I don't know about...

Reply:
I'll change it the way you suggested :smiley: :+1: It might break something, but I think we should find that as quick as possible, because it just doesn't make sense that the raw `superclass` string property and the erasure of the generic signature don't match. I would rather find that out and then explicitly document a workaround / edge case instead of keeping the old behavior for `getRawSuperclass()` at the cost of maintainability / inconsistencies later.